src/mintpy/save_hdfeos5.py

```diff
@@ -103,7 +104,15 @@ def metadata_mintpy2unavco(meta_in, dateList, geom_file):
     unavco_meta['beam_swath'] = int(meta.get('beam_swath', '0'))
 
     # relative_orbit, or track number
-    unavco_meta['relative_orbit'] = int(meta['relative_orbit'])
+    try:
+        unavco_meta['relative_orbit'] = int(meta['relative_orbit'])
+
+    except ValueError:
+        try:
+            geom_meta = readfile.read_attribute(geom_file)         # FA 12/22: for TSX meta does not have relative_orbit attribute
+            unavco_meta['relative_orbit'] = int(geom_meta['unavco.relative_orbit'])
+        except ValueError:
+            print('Missing required attribute: relative_orbit')
```

Comment:
I don't see the motivation of this change here: time-series file is supposed to have more metadata than geometry files, always. Isn't that the case for your dataset?

FYI, the `unavco.relative_orbit` should had been translated into `relative_orbit` [here](https://github.com/insarlab/MintPy/blob/c7400f97ffd370b0140906cbf3925ead6de10fc1/src/mintpy/save_hdfeos5.py#L80-L87).

Reply:
It could be that this is required for MiaplPy and that `load_slc_geometry.py` of MiaplPy does not use this MintPy function. `relative_orbit` in `slcStack.h5` is currently none as you can see below. Maybe we should make a note how how this could be improved.

```
MiamiTsxSMD14/miaplpy/inputs[1014] info.py slcStack.h5 
******************** Basic File Info ************************
file name: /scratch/05861/tg851601/MiamiTsxSMD14/miaplpy/inputs/slcStack.h5
file type: timeseries
coordinates : RADAR

******************** Date Stat Info *************************
Start Date: 20170923
End   Date: 20210618
Number of dates  : 100
STD of datetimes : 1.08 years

******************** HDF5 File Structure ********************
Attributes in / level:
  ALOOKS                   1
  ANTENNA_SIDE             -1
  AZIMUTH_PIXEL_SIZE       2.1495932845300736
  BANDS                    1
  CENTER_LINE_UTC          40691.0
  DATA_TYPE                complex64
  DATE                     20200529
  EARTH_RADIUS             6348107.616403343
  FILE_LENGTH              7046
  FILE_PATH                /scratch/05861/tg851601/MiamiTsxSMD14/merged/SLC/20200529/20200529.slc
  FILE_TYPE                timeseries
  HEADING                  -168.4374330199127
  HEIGHT                   511035.8120019732
  INTERLEAVE               BIP
  LAT_REF1                 25.921521074257434
  LAT_REF2                 25.94302544748516
  LAT_REF3                 25.797651852239053
  LAT_REF4                 25.819170744512242
  LENGTH                   7046
  LON_REF1                 -80.02806403183072
  LON_REF2                 -80.16693071534473
  LON_REF3                 -80.05174311730326
  LON_REF4                 -80.19047876057562
  NCORRLOOKS               0.8178098213569981
  PLATFORM                 Tsx
  POLARIZATION             HH
  PRF                      3577.0218098958335
  PROCESSOR                isce
  PROJECT_NAME             MiamiTsxSMD14
  P_BASELINE_BOTTOM_HDR    -460.76757280546883
  P_BASELINE_TOP_HDR       -460.0380850366407
  RANGE_PIXEL_SIZE         1.3641053359701238
  RLOOKS                   1
  STARTING_RANGE           681478.0615665995
  SUBSET_XMAX              7168
  SUBSET_XMIN              0
  SUBSET_YMAX              7046
  SUBSET_YMIN              0
  UNIT                     i
  WAVELENGTH               0.03106658115133631
  WIDTH                    7168
  XMAX                     7167
  YMAX                     7045
  access_mode              read
  altitude                 511035.8120019732
  azimuthPixelSize         2.1495932845300736
  azimuthResolution        2.392
  beam_mode                SM
  byte_order               l
  data_type                complex64
  description              ['Single look complex image.']
  earthRadius              6348107.616403343
  extra_file_name          /scratch/05861/tg851601/MiamiTsxSMD14/merged/SLC/20170923/20170923.slc.vrt
  family                   slcimage
  file_name                /scratch/05861/tg851601/MiamiTsxSMD14/merged/SLC/20170923/20170923.slc
  image_type               slc
  isce_version             Release: 2.6.1, svn-, 20220811. Current: svn-.
  length                   7046
  name                     slcimage_name
  number_bands             1
  orbitNumber              56649
  polarization             HH
  prf                      3577.0218098958335
  radarWavelength          0.03106658115133631
  rangePixelSize           1.3641053359701238
  rangeResolution          1.49896229
  relative_orbit           None
  satelliteSpeed           7689.142061169693
  scheme                   BIP
  startUTC                 2017-09-01 11:18:06.651931
  startingRange            681478.0615665995
  stopUTC                  2017-09-01 11:18:17.196737
  trackNumber              None
  width                    7168
  xmax                     7168
  xmin                     0

HDF5 dataset "/bperp               ": shape=(100,)              , dtype=float32   , compression=None
HDF5 dataset "/date                ": shape=(100,)              , dtype=|S8       , compression=None
HDF5 dataset "/slc                 ": shape=(100, 7046, 7168)   , dtype=complex64 , compression=None
  MODIFICATION_TIME    1672180816.6942618
```

Reply:
I don't like this change because it makes the code messy. I suggest grabbing the valid metadata, such as `relative_orbit`, from the geometry file [here](https://github.com/insarlab/MintPy/blob/4e88f085cf6e94d36924599fab05935f0c806b9f/src/mintpy/save_hdfeos5.py#L50), if that has more metadata than the time-series file. 

IMO, both files should have these types of metadata consistent in the miaplpy stage.